x-pack/plugins/ml/server/models/results_service/get_partition_fields_values.ts

```diff
@@ -82,68 +133,85 @@ export const getPartitionFieldsValuesFactory = ({ asInternalUser }: IScopedClust
    * @param criteriaFields - key - value pairs of the term field, e.g. { detector_index: 0 }
    * @param earliestMs
    * @param latestMs
+   * @param fieldsConfig
    */
   async function getPartitionFieldsValues(
     jobId: string,
     searchTerm: SearchTerm = {},
     criteriaFields: CriteriaField[],
     earliestMs: number,
-    latestMs: number
+    latestMs: number,
+    fieldsConfig: FieldsConfig = {}
   ) {
     const { body: jobsResponse } = await asInternalUser.ml.getJobs({ job_id: jobId });
     if (jobsResponse.count === 0 || jobsResponse.jobs === undefined) {
       throw Boom.notFound(`Job with the id "${jobId}" not found`);
     }
 
-    const job = jobsResponse.jobs[0];
+    const job: Job = jobsResponse.jobs[0];
 
     const isModelPlotEnabled = job?.model_plot_config?.enabled;
+    const isAnomalousOnly = (Object.entries(fieldsConfig) as Array<[string, FieldConfig]>).some(
+      ([k, v]) => {
+        return !!v?.anomalousOnly;
+      }
+    );
+    const isModelPlotSearch = !!(isModelPlotEnabled && !isAnomalousOnly);
 
-    const { body } = await asInternalUser.search({
-      index: ML_RESULTS_INDEX_PATTERN,
-      size: 0,
-      body: {
-        query: {
-          bool: {
-            filter: [
-              ...criteriaFields.map(({ fieldName, fieldValue }) => {
-                return {
-                  term: {
-                    [fieldName]: fieldValue,
-                  },
-                };
-              }),
-              {
+    const requestBody = {
+      query: {
+        bool: {
+          filter: [
+            ...criteriaFields.map(({ fieldName, fieldValue }) => {
+              return {
                 term: {
-                  job_id: jobId,
+                  [fieldName]: fieldValue,
                 },
+              };
+            }),
+            {
+              term: {
+                job_id: jobId,
               },
-              {
-                range: {
-                  timestamp: {
-                    gte: earliestMs,
-                    lte: latestMs,
-                    format: 'epoch_millis',
+            },
+            // Remove the time filter in case model plot is not enabled
+            // and not only anomalous records have been requested.
+            // It includes the records that occurred as anomalies historically
+            ...(!isModelPlotEnabled && !isAnomalousOnly
```

Comment:
Isn't this back to front? If isAnomalousOnly is false, i.e. if the user doesn't want to restrict, then you *include* a filter on time range. In general, this logic feels complicated because it is kind of a double negative plus a ternary. I would be inclined to define a variable for this, something like `const searchAllTime = !!(!isModelPlotEnabled && !isAnomalousOnly)`.

Reply:
thanks for spotting this! fixed in https://github.com/elastic/kibana/pull/81510/commits/faeaf2f10829f5dc2034bdcbce88cf01cd6c9b2d